### PR TITLE
Integrate SwanLab for offline/online experiment tracking for Accelerate

### PR DESCRIPTION
**Pull request by ShaohonChen**
Frame 4478

# What does this PR do?

This PR introduces [SwanLab](https://github.com/SwanHubX/SwanLab), a lightweight open-source experiment tracking tool, as a new logging option for the training framework. The integration provides both online and offline tracking capabilities, along with a local dashboard for visualizing results.

SwanLab has previously supported:
- Tracking via Transformers' `report_to` parameter ([documentation](https://huggingface.co/docs/transformers/v4.52.3/en/main_classes/callback#transformers.integrations.SwanLabCallback))
- The Accelerate training framework through external callbacks ([documentation](https://docs.swanlab.cn/en/guide_cloud/integration/integration-huggingface-accelerate.html))

We've received numerous requests from the community to add native Accelerate support (see [here](https://github.com/SwanHubX/SwanLab/issues/1014)), and we're excited to officially integrate with this excellent project to provide a more seamless experience for developers. This integration is particularly valuable for users in regions with limited network connectivity (such as China), offering them robust experiment tracking capabilities.

image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),  Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. (see [here](https://github.com/SwanHubX/SwanLab/issues/1014))
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [X] Did you write any new necessary tests? (I don't see any tests for any of the callbacks but please let me know if I missed them somewhere. )

## Who can review?

@SunMarc  I noticed that you recently reviewed some related PRs—would you mind helping review my PR as well? Thank you! (I believe you also reviewed the Hugging Face Transformers integration previously—looking forward to collaborating again! 😄)

---

### Usage guidline

**Step 0: Set Up Accelerate and example environment**

Following the **accelerate** official cv example (pet image classification task):

```bash
# prepare code and environments
git clone https://github.com/huggingface/accelerate
cd accelerate
pip install -e .
pip install timm     # use in example
```

**Step 1: Set Up SwanLab Online Tracking**

Install:

```bash
pip install swanlab
```

To use SwanLab's online tracking, log in to the [SwanLab website](https://swanlab.cn/) and obtain your API key from the [Settings page](https://swanlab.cn/space/~/settings). Then, authenticate using the following command:

```bash
swanlab login
```

If you prefer offline mode, skip this step and install local dashboard:

```bash
pip install swanlab[dashboard]
```

**Step 2: download *Oxford-IIT Pet Dataset* used in example code**

You can find download link [here](https://www.robots.ox.ac.uk/~vgg/data/pets/)

**Step 3: run offical example script in accelerate projects**

```bash
python examples/complete_cv_example.py  --data_dir <DOWNLOAD DATA PATH> --with_tracking
```

Screenshot 2025-06-03 at 21 12 01

Screenshot 2025-06-03 at 21 06 48

Screenshot 2025-06-03 at 21 12 33

[visualization demo here](https://swanlab.cn/@ShaohonChen/complete_cv_example/runs/5kuj38hkwytm0oob9uis2/chart)

Since my server is offline, I changed the `pretrain` parameter to `false` in the `create_model` code to avoid downloading the model online, which led to very poor accuracy after just 3 epochs 😂.

### test suite passes

Screenshot 2025-06-03 at 21 01 04

Since my AI training server couldn't connect to Hugging Face, some tests failed during the automated testing process.😭

### CODE REVIEW

**Comment by ShaohonChen** [comment]
Hi, we're currently updating the local log storage format for SwanLab, and a new version will be released within 2 days. We'll also complete the corresponding test cases for the new version!

**Comment by ShaohonChen** [comment]
I’ve added some new test cases. Would you mind helping me review them when you're available? Thanks a lot for your time! @SunMarc

**Comment by ShaohonChen** [comment]
Thanks for the review and merge! 🤗
